src/stereo_undistort.cpp

```diff
@@ -0,0 +1,265 @@
+      ROS_ERROR("%s", e.what());
+      return;
+    }
+  }
+}
+
+void StereoUndistort::sendCameraInfo(const std_msgs::Header& header,
+                                     const bool left, const bool input) {
+  sensor_msgs::CameraInfo camera_info;
+  camera_info.header = header;
+  try {
+    stereo_camera_parameters_ptr_->generateCameraInfoMessage(left, input,
+                                                             &camera_info);
+  } catch (std::runtime_error e) {
+    ROS_ERROR("%s", e.what());
+    return;
+  }
+
+  if (rename_radtan_plumb_bob_ && camera_info.distortion_model == "radtan") {
+    camera_info.distortion_model = "plumb_bob";
+  }
+
+  if (left) {
+    if (input) {
+      left_camera_info_input_pub_.publish(camera_info);
+    } else {
+      left_camera_info_output_pub_.publish(camera_info);
+    }
+  } else {
+    if (input) {
+      right_camera_info_input_pub_.publish(camera_info);
+    } else {
+      right_camera_info_output_pub_.publish(camera_info);
+    }
+  }
+}
+
+void StereoUndistort::processAndSendImage(
+    const sensor_msgs::ImageConstPtr& image_msg_in, const bool left) {
+  cv_bridge::CvImageConstPtr image_in_ptr =
+      cv_bridge::toCvShare(image_msg_in, output_image_type_);
+  cv_bridge::CvImagePtr image_out_ptr(
+      new cv_bridge::CvImage(image_in_ptr->header, image_in_ptr->encoding));
+
+  updateUndistorter(left);
+
+  image_out_ptr->header.frame_id = output_frame_;
+
+  if (left) {
+    left_undistorter_ptr_->undistortImage(image_in_ptr->image,
+                                          &(image_out_ptr->image));
+    left_image_pub_.publish(*(image_out_ptr->toImageMsg()));
+
+    if (publish_tf_) {
+      Eigen::Matrix4d T =
+          stereo_camera_parameters_ptr_->getLeft()
+              .getInputPtr()
+              ->T()
+              .inverse() *
+          stereo_camera_parameters_ptr_->getRight().getOutputPtr()->T();
+
+      tf::Matrix3x3 R_ros;
+      tf::Vector3 p_ros;
+      tf::matrixEigenToTF(T.topLeftCorner<3, 3>(), R_ros);
+      tf::vectorEigenToTF(T.topRightCorner<3, 1>(), p_ros);
+      tf::Transform(R_ros, p_ros);
+
+      br_.sendTransform(tf::StampedTransform(
+          tf::Transform(R_ros, p_ros), image_out_ptr->header.stamp,
+          image_in_ptr->header.frame_id, output_frame_));
+    }
+  } else {
+    right_undistorter_ptr_->undistortImage(image_in_ptr->image,
+                                           &(image_out_ptr->image));
+    right_image_pub_.publish(*(image_out_ptr->toImageMsg()));
+  }
+}
+
+void StereoUndistort::imagesCallback(
+    const sensor_msgs::ImageConstPtr& left_image_msg_in,
+    const sensor_msgs::ImageConstPtr& right_image_msg_in) {
+  if (!stereo_camera_parameters_ptr_->valid()) {
+    ROS_ERROR("Camera parameters invalid, undistortion failed");
+    return;
+  }
+
+  if (++frame_counter_ < process_every_nth_frame_) {
+    return;
+  }
+  frame_counter_ = 0;
+
+  processAndSendImage(left_image_msg_in, true);
```

Comment:
Would be nicer to have these booleans (and below) be named consts so we know what they mean when reading the code...